src/AssignmentsStorage.php

```diff
@@ -0,0 +1,209 @@
+        $this->load();
+    }
+
+    public function getAssignments(): array
+    {
+        return $this->assignments;
+    }
+
+    public function getUserAssignments(string $userId): array
+    {
+        return $this->assignments[$userId] ?? [];
+    }
+
+    public function getUserAssignmentByName(string $userId, string $name): ?Assignment
+    {
+        return $this->getUserAssignments($userId)[$name] ?? null;
+    }
+
+    public function addAssignment(string $userId, Item $item): void
+    {
+        $this->assignments[$userId][$item->getName()] = new Assignment($userId, $item->getName(), time());
+        $this->saveAssignments();
+    }
+
+    public function assignmentExist(string $name): bool
+    {
+        foreach ($this->getAssignments() as $assignmentInfo) {
+            foreach ($assignmentInfo as $itemName => $assignment) {
+                if ($itemName === $name) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    public function updateAssignmentsForItemName(string $name, Item $item): void
+    {
+        foreach ($this->assignments as &$assignments) {
+            if (isset($assignments[$name])) {
+                $assignments[$item->getName()] = $assignments[$name]->withItemName($item->getName());
+                unset($assignments[$name]);
+            }
+        }
+        $this->saveAssignments();
+    }
+
+    public function removeAssignment(string $userId, Item $item): void
+    {
+        unset($this->assignments[$userId][$item->getName()]);
+        $this->saveAssignments();
+    }
+
+    public function removeAllAssignments(string $userId): void
+    {
+        $this->assignments[$userId] = [];
+        $this->saveAssignments();
+    }
+
+    public function removeAssignmentsFromItem(Item $item): void
+    {
+        $this->clearAssignmentsFromItem($item);
+        $this->saveAssignments();
+    }
+
+    public function clearAssignments(): void
+    {
+        $this->assignments = [];
+        $this->saveAssignments();
+    }
+
+    /**
+     * Loads authorization data from persistent storage.
+     */
+    private function load(): void
+    {
+        $this->loadAssignments();
+    }
+
+    private function loadAssignments(): void
+    {
+        $assignments = $this->loadFromFile($this->assignmentFile);
+        $assignmentsMtime = @filemtime($this->assignmentFile);
+        foreach ($assignments as $userId => $roles) {
+            foreach ($roles as $role) {
+                $this->assignments[$userId][$role] = new Assignment((string)$userId, $role, $assignmentsMtime);
+            }
+        }
+    }
+
+    /**
+     * Loads the authorization data from a PHP script file.
+     *
+     * @param string $file The file path.
+     *
+     * @return array The authorization data.
+     *
+     * @see saveToFile()
+     */
+    private function loadFromFile(string $file): array
```

Comment:
Common code from stroges can move to separate internal class or to abstract class.